src/modules/appointment/appointment.service.ts

```diff
@@ -0,0 +1,111 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository, SelectQueryBuilder } from 'typeorm';
+import DoctorService from 'modules/doctor/doctor.service';
+import PatientService from 'modules/patient/patient.service';
+import Appointment from './entity/appointment.entity';
+import CreateAppointmentDto from './dto/create-appointment.dto';
+
+@Injectable()
+export default class AppointmentService {
+  constructor(
+    @InjectRepository(Appointment)
+    private appointmentRepository: Repository<Appointment>,
+    private readonly doctorService: DoctorService,
+    private readonly patientService: PatientService,
+  ) {}
+
+  async createAppointment(
+    appointment: CreateAppointmentDto,
+  ): Promise<Appointment> {
+    try {
+      const { localDoctorId, remoteDoctorId, patientId, ...rest } = appointment;
+
+      const [patient, localDoctor, remoteDoctor] = await Promise.all([
+        this.patientService.getPatientById(patientId),
+        this.doctorService.getDoctorByID(localDoctorId),
+        this.doctorService.getDoctorByID(remoteDoctorId),
+      ]);
```

Comment:
why we need 3 query, to. make insert to Appointment we need id,not full object
find other way

Reply:
rewrited and inserted only ids on appointment